editor.planx.uk/src/pages/Pay/MakePayment.tsx

```diff
@@ -44,138 +50,190 @@ enum PaymentState {
+      setErrorMessage("Failed to fetch payment details");
+    }
+
+    if (responseData) resolvePaymentResponse(responseData);
+    setIsLoading(false);
+    switch (computePaymentState(responseData)) {
+      case PaymentState.NotStarted:
+        setState(States.Ready);
+        break;
+      case PaymentState.Pending:
+        setState(States.ReadyToRetry);
+        break;
+      case PaymentState.Failed:
+        setState(States.ReadyToRetry);
+        setPayment(undefined);
+        break;
+      case PaymentState.Completed:
+        setState(States.Finished);
+        break;
+    }
   };
 
-  const resolvePaymentResponse = (responseData: GovUKPayment) => {
+  const resolvePaymentResponse = (responseData: GovUKPayment): GovUKPayment => {
     if (!responseData?.state?.status)
       throw new Error("Corrupted response from GOV.UK");
-    let payment: GovUKPayment = {
+    const resolvedPayment: GovUKPayment = {
       ...responseData,
       amount: toDecimal(responseData.amount),
     };
-    setPayment(payment);
+    setPayment(resolvedPayment);
+    // useState is async, so we also pass the resolved value to the chained promise
+    return resolvedPayment;
   };
 
   const readyAction = async () => {
-    isLoading(true);
+    setIsLoading(true);
     if (payment && currentState === States.ReadyToRetry) {
       redirectToGovPay(payment);
     } else {
       await startNewPayment(paymentRequestId)
         .then(resolvePaymentResponse)
-        .then(() => redirectToGovPay(payment))
+        .then(redirectToGovPay)
         .catch(logger.notify);
     }
   };
 
-  return (
-    <Box>
-      <Typography variant="h1" gutterBottom>
+  const Header = () =>
+    currentState === States.Finished ? (
+      <Banner
+        heading="Payment received"
+        color={{
+          background: lighten(theme.palette.success.main, 0.9),
+          text: "black",
+        }}
+      >
+        <Typography pt={2} variant="body2">
+          Thanks for making your payment. We'll send you a confirmation email.
+        </Typography>
+      </Banner>
+    ) : (
+      <Typography maxWidth="md" variant="h1" pt={5} gutterBottom>
         Pay for your application
       </Typography>
-      {/* TODO - this is just a placeholder */}
-      <table>
-        <tr>
-          <th>Application type</th>
-          <td>...</td>
-        </tr>
-        <tr>
-          <th>Fee</th>
-          <td>{formattedPriceWithCurrencySymbol(paymentAmount)}</td>
-        </tr>
-        <tr>
-          <th>Address</th>
-          <td>...</td>
-        </tr>
-        <tr>
-          <th>Project type</th>
-          <td>...</td>
-        </tr>
-      </table>
-      <Typography variant="body1">
+    );
+
+  const PaymentDetails = () => {
+    const data = [
+      { term: "Application type", details: flowName },
+      {
+        term: "Fee",
+        details: formattedPriceWithCurrencySymbol(toDecimal(paymentAmount)),
+      },
+      {
+        term: "Address",
```

Comment:
nit: this is referred to as "Property address" on the Confirmation page, rather than just "Address". Happy to bookmark and revisit in a future finalisation of copy (I'd guess we want to double check consistency of Notify templates too)